content/docs/rfcs/41/README.md

```diff
@@ -0,0 +1,85 @@
+---
+slug: 41
+title: 41/WAKU2-RLN-CONTRACT
+name: Rate Limiting Nullifier Contract
+status: raw
+category: RLN
+tags: 17/WAKU-RLN-RELAY
+editor: Keshav Gupta <keshav@status.im>
+contributors:
+---
+
+## Tags
+
+Currently identified tags comprise
+
+* `17/WAKU-RLN-RELAY` protocol is an extension of `11/WAKU-RELAY` which additionally provides spam protection using Rate Limiting Nullifiers (RLN).
+
+
+# Abstract
+The following standard allows for the implementation of a standard API for Rate Limiting Nullifier Contract that manages membership of the participants in a group.
+
+
+# Background / Rationale / Motivation
+
+This contract serves as the basis for the implementation of RLN membership. New members can be registered by depositing a stake of valuable funds.
+They can also be slashed if they misbehave like spamming in a peer to peer pseudo-anonymous messaging network.
+
+# Theory / Semantics
+
+As explained in [17/WAKU-RLN-RELAY](https://rfc.vac.dev/spec/17/), all peers who want to publish the message in the spam protected peer to peer network MUST register for membership to the RLN group. This is done via the `register` function of the RLN membership contract. If a registered member of the RLN group spams messages in the network then its secret key get exposed which can be used to slash its membership from the group. This is done by calling the `withdraw` function of the RLN membership contract, which removes the member from the merkle tree thereby revoking its rights to send any messages.
+
+# Wire Format Specification / Syntax
+
+### Constructor
+
+* `constructor(uint256 membershipDeposit, uint256 depth, address _poseidonHasher) public`
+    * `membershipDeposit` sets the amount of ETH that must be deposited to `register` to the RLN group.
+    * `depth` is the depth of the merkle tree that determines the maximum number of members to the RLN group.
```

Comment:
The Merkle tree has not been introduced before in this RFC. It needs to be explained in the "background" section.